docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,261 @@
 
-```bash
-oras backup --output ./mirror  registry.k8s.io/kube-apiserver-arm64:v1.31.0 registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+**Syntax:**
+```console
+oras restore --flags <registry>/<repository>[:<ref1>[,<ref2>...]] [...]
 ```
 
-It is mandatory to specify `--output` argument with the destination.
-The source artifacts may be read from different registries although the example reads artifacts from one registry.
-If no reference tag or digest is specified, the entire repository will be copied.
+**Output:**
+Artifacts are uploaded to the target registry/registries as specified.
 
-The output directory structure is a single OCI layout containing all of the artifacts.
-Each artifact in the output OCI layout will be tagged with the name of source.
-For example, the `registry.k8s.io/kube-apiserver-arm64:v1.31.0` artifact will be tagged `registry.k8s.io/kube-apiserver-arm64:v1.31.0`.
+**New flags:**
 
-```bash
-% oras repo tags --oci-layout ./mirror
-registry.k8s.io/kube-apiserver-arm64:v1.31.0
-registry.k8s.io/kube-apiserver-arm64:v1.32.0
-registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+- `--input <path>`: Required. Restore from a folder or archive file to registry.
+- `--exclude-referrers`: Restore the image from backup excluding referrers.
+
+**Common flags:**
+
+* `--concurrency <int>`: Number of parallel upload operations. Default: `3`.
+* `--plain-http`: Allow insecure connections to registry without SSL check.
+* `--insecure`: Allow connections to registries without valid TLS certificates.
+* `--registry-config <path>`: Path to the authentication configuration file for the registry.
+* `--username <string>`: Username for authenticating to the registry.
+* `--password <string>`: Password for authenticating to the registry.
+* `--password-stdin`: Read password from stdin.
+* `--identity-token <string>`: Use bearer token for authentication.
+* `--identity-token-stdin`: Read identity token from stdin.
+* `--ca-file <path>`: Path to custom CA certificate file.
+* `--cert-file <path>`: Path to client TLS certificate file.
+* `--key-file <path>`: Path to client private key file.
+* `--resolve <host:port:address[:address_port]>`: Customized DNS for registry.
+* `--debug`: Output debug logs (implies `--no-tty`).
+* `--distribution-spec string`: [Preview] set OCI distribution spec version and API option for target. Options: v1.1-referrers-tag, v1.1-referrers-api
+* `--no-tty`: Disable progress bars.
+
+### User Experience in the CLI
+
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
+
+#### Offline Snapshot for Air-Gapped Environments
+
+Create a snapshot of a sample image `registry-a.k8s.io/kube-apiserver:v1` and its referrer (e.g. signature) for an air-gapped environment:
+
+```console
+oras backup registry-a.k8s.io/kube-apiserver:v1 --include-referrers --output airgap-snapshot.tar
```

Comment:
It is better to provide a full repo dump scenario instead of a single image dump. This is because a single image dump can be easily done by a single `oras copy` command that cannot highlight the necessity of `oras backup`.

Reply:
I noticed that you have the section "Backup an Entire Repository to a Tarball and Restore to Another Registry" below. Do you want to bring that section to the top and remove the "Offline Snapshot for Air-Gapped Environments" section?

Reply:
I moved the section "Backup an Entire Repository to a Tarball and Restore to Another Registry" to the bottom. Backup of a single is still a necessary example as it aligns with the scenario 1 and scenario 2.